content/en/blog/_posts/2022-12-16-statefulset-migration.md

```diff
@@ -0,0 +1,206 @@
+7743661f60b6b17b5c71d083260419588b4f2451 10.104.0.16:6379@16379 slave 2ce30362c188aabc06f3eee5d92892d95b1da5c3 0 1669764410000 3 connected                                                                                             
+961f35e37c4eea507cfe12f96e3bfd694b9c21d4 10.104.0.18:6379@16379 slave a8765caed08f3e185cef22bd09edf409dc2bcc61 0 1669764411000 1 connected                                                                                                             
+7136e37d8864db983f334b85d2b094be47c830e5 10.104.0.15:6379@16379 slave 2cff613d763b22c180cd40668da8e452edef3fc8 0 1669764412595 2 connected                                                                                                                    
+a8765caed08f3e185cef22bd09edf409dc2bcc61 10.104.0.19:6379@16379 master - 0 1669764411592 1 connected 0-5460                                                                                                                                                   
+2cff613d763b22c180cd40668da8e452edef3fc8 10.104.0.17:6379@16379 master - 0 1669764410000 2 connected 5461-10922
+```
+
+5. On `destination`, deploy Redis with zero replicas.
+
+```
+helm install redis --namespace kep-3335 \
+  bitnami/redis-cluster \
+  --set persistence.size=1Gi \
+  --set cluster.nodes=0 \
+  --set redis.extraEnvVars\[0\].name=REDIS_NODES,redis.extraEnvVars\[0\].value="redis-redis-cluster-headless.kep-3335.svc.cluster.local" \
+  --set existingSecret=redis-redis-cluster
+```
+
+6. Scale down replica `redis-redis-cluster-5` in the source cluster.
+
+```
+kubectl patch sts redis-redis-cluster -p '{"spec": {"replicas": 5}}'
+```
+
+7. Migrate dependencies from `source` to `destination`.
+
+The following commands copy resources from `source` to `destionation`. Details
+that are not relevant in `destination` cluster are removed (eg: `uid`,
+`resourceVersion`, `status`).
+
+Note: For the PV/PVC, this procedure only works if the underlying storage system
+      that your `StorageClass` uses can support copying existing PVs into a
+      new cluster. Storage that is associated with a specific node or topology
+      may not be supported. Additionally, some storage systems may store
+      addtional metadata about volumes outside of a PV object, and may require
+      a more specialized sequence to import a volume.
+
+Source Cluster
+
+```
+kubectl get pvc redis-data-redis-redis-cluster-5 -o yaml | yq 'del(.metadata.uid, .metadata.resourceVersion, .metadata.annotations, .metadata.finalizers, .status)' > /tmp/pvc-redis-data-redis-redis-cluster-5.yaml
+kubectl get pv $(yq '.spec.volumeName' /tmp/pvc-redis-data-redis-redis-cluster-5.yaml) -o yaml | yq 'del(.metadata.uid, .metadata.resourceVersion, .metadata.annotations, .metadata.finalizers, .spec.claimRef, .status)' > /tmp/pv-redis-data-redis-redis-cluster-5.yaml
```

Comment:
Should we set the PV reclaimPolicy to retain to avoid accidental disk deletion in the source cluster?

Reply:
Good idea, added a note to a reference guide on how to do so.